src/api/libcellml/enums.h

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <memory>
+
+namespace libcellml {
+
+/**
+ * @brief The ItemType enum class.
+ *
+ */
+enum class ItemType
```

Comment:
I'm not a fan of the name for this enumeration.  I think it should be `ReferenceType` or `ReferenceCellMLType`.  With documentation stating that it is a reference that identifies a CellML 2.0 specification element information item.

I can't decide if the `CellML` is needed in the name or not.

Reply:
I don't think that CellML can be in the name.  It needs to have an XML class, and that's outside CellML ... I also don't think "reference" fits either.  It's the type of thing which is stored in the issue.  Whether that's a CellML element or something else, I thought that "item" was pretty broad ...

Reply:
I don't think `XML` needs to be in this enum.  To me it is telling us what sort of thing is stored in the issue thus `XML` shouldn't be here as it is not something that is stored in the issue.  For me this has all become very muddled up and that the `ReferenceType` should describe the reference that is being held be it a `Variable`, `Component`, `VariablePair`, `UnitItem`, etc.  The `ReferenceRule` would specify what is wrong which is where the `XML` comes in.  `ReferenceRule` already has an `UNDEFINED` value but it could also have `NOT_SPECIFIED` to indicate that the reference rule that is broken is not being specified.

Reply:
It's there because the old "cause" and "item type" are 99% the same ... but if you want to be able to report a "cause" being something related to XML in the `Parser` then you need this category.  I think that a not specified entry isn't helpful, especially when we actually do know where the problem is!

Reply:
I'm thinking that the `cause` should go and we simply use `referenceRule` to do that job.

Reply:
Yeah, except that these `Issue` items need to be relevant to all of the `Logger` types, not just the `Validator`.  Is there another name that doesn't imply it's validation related?

Reply:
The `Validator` is restricted to the issues that it can raise.  Which is a subset of the reference rules.  We already have reference rules that are not available to the validation.  This was always the vision, the `Validator` validates against the specification and uses reference rules to identify exactly where in the specification the error arrises from.  Reference rules would be extended for other reference material.  For an issue that doesn't come from a defined reference we have `UNDEFINED`.

Reply:
What's the situation with this?

Reply:
XML is gone now.